src/blocks/content-helper/content-helper-provider.ts

```diff
@@ -36,33 +65,39 @@ class ContentHelperProvider {
 		const tag: Taxonomy = select( 'core' ).getEntityRecord( 'taxonomy', 'post_tag', tagIds[ 0 ] );
 
 		// Create API query.
-		const fetchQueryResult = this.buildFetchDataQuery( author, category, tag );
-		if ( fetchQueryResult.query === null ) {
-			return Promise.reject( fetchQueryResult.message );
+		const apiQuery = this.buildRelatedTopPostsApiQuery( author, category, tag );
+		if ( apiQuery.query === null ) {
+			return Promise.reject( apiQuery.message );
 		}
 
 		// Fetch results from API and set the Content Helper's message.
-		const data = await this.fetchData( fetchQueryResult );
-		let message = `${ __( 'Top-performing posts', 'wp-parsely' ) } ${ fetchQueryResult.message }.`;
+		const data = await this.fetchRelatedTopPostsFromWpEndpoint( apiQuery );
+		let message = `${ __( 'Top-performing posts', 'wp-parsely' ) } ${ apiQuery.message }.`;
 
 		if ( typeof data === 'string' ) {
 			return { message: data, posts: [] };
 		}
 
 		if ( data.length === 0 ) {
-			message = `${ __( 'The Parse.ly API did not return any results for top-performing posts', 'wp-parsely' ) } ${ fetchQueryResult.message }.`;
+			message = `${ __( 'The Parse.ly API did not return any results for top-performing posts', 'wp-parsely' ) } ${ apiQuery.message }.`;
 		}
 
 		return { message, posts: data };
 	}
 
-	private static async fetchData( fetchDataQueryResult: BuildFetchDataQueryResult ): Promise<SuggestedPost[] | string> {
+	/**
+	 * Fetches the related top-performing posts data from the WordPress REST API.
+	 *
+	 * @param {RelatedTopPostsApiQuery} fetchDataQueryResult
+	 * @return {Promise<Array<RelatedTopPostData>>} Array of fetched posts or empty array.
+	 */
+	private static async fetchRelatedTopPostsFromWpEndpoint( fetchDataQueryResult: RelatedTopPostsApiQuery ): Promise<RelatedTopPostData[] | string> {
```

Comment:
`fetchDataQueryResult` seems like an odd name for this input argument

Reply:
Indeed!